ImmuDB4Net/ImmuClientSync.cs

```diff
@@ -0,0 +1,1364 @@
+/*
+Copyright 2022 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+namespace ImmuDB;
+
+using Google.Protobuf;
+using Google.Protobuf.Collections;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+using ImmuDB.Crypto;
+using ImmuDB.Exceptions;
+using ImmudbProxy;
+using Org.BouncyCastle.Crypto;
+
+public partial class ImmuClientSync
```

Comment:
Suggestion: There are many code parts shared between sync and async client (like validation of proofs) - we could have some internal common class that both ImmuClientSync and ImmuClientAsync derive from to avoid code duplication.